### PR TITLE
Release `0.19.2`

### PR DESCRIPTION
**Pull request by SajidAlamQB**
## Description

Minor release.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Review by astrojuanlu — APPROVED** [review]
🚀

File: RELEASE.md [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
 ## Breaking changes to the API
-* Added logging about not using async mode in `SequentiallRunner` and `ParallelRunner`.
 
 ## Documentation changes
-* Added documentation about `bootstrap_project` and `configure_project`.
-* Added documentation about `kedro run` and hook execution order.
 
 ## Community contributions
 
+# Upcoming Release 0.19.2
```
Comment (merelcht) [inline RELEASE.md:13]:
```suggestion
# Release 0.19.2
```

**Review by merelcht — APPROVED** [review]

**Comment by SajidAlamQB** [comment]
CI will complain for  as we are now using the `kedro-starters` `tag` version over the `main` branch. We will have to make some adjustments to CI so its always using main.

**Comment by SajidAlamQB** [comment]
Release is blocked by this PR: https://github.com/kedro-org/kedro/pull/3518

File: CITATION.cff [hunk 2]
```diff
@@ -51,6 +51,6 @@ authors:
 - family-names: Theisen
   given-names: Merel
 title: Kedro
-version: 0.19.1
-date-released: 2023-12-13
+version: 0.19.2
+date-released: 2024-01-16
```
Comment (merelcht) [inline CITATION.cff:55]: Don't forget to update this when we do the release on Monday!

**Pull request merged**